std/zip.d

```diff
@@ -823,6 +825,78 @@ final class ZipArchive
     {
         data[i .. i + 8] = nativeToLittleEndian(ul);
     }
+
+    /* ============== for detecting overlaps =============== */
+
+private:
+
+    // defines a segment of the zip file, including start, excluding end
+    struct Segment
+    {
+        uint start;
+        uint end;
+    }
+
+    // removes Segment start .. end from _segs
+    // throws zipException if start .. end is not completely available in _segs;
+    void removeSegment(uint start, uint end) pure @safe
```

Comment:
Perhaps add a contract here, e.g. `in { assert(start > end); }`.

Reply:
Good idea. :-)